### PR TITLE
Replace some chat messages for guns with balloon alerts

### PR DESCRIPTION
**Pull request by msgerbs**
## About The Pull Request
Guns have a lot of related messages that get sent to chat. Some of these are pretty damn important to be able to see a fight, even though they don't need to stick around. That makes them ideal candidates to use balloon alerts instead.

image
image
image

I also removed an entirely unneeded duplicate proc for burst fire that just did exactly the same as the parent proc except slightly shittier.

## Why It's Good For The Game
Balloon alerts are pretty cool and help reduce chat clogging, we should use them wherever they make sense and I think they make sense here.

## Changelog

:cl: VexingRaven
qol: Many gun-related chat messages are now balloon alerts instead
spellcheck: The bow no longer says you loaded it with bullets
/:cl:

### CODE REVIEW

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by SinfulBliss** [comment]
Is this only visible to the player or to everyone? Things like stealth loading a 357 in your bag showing up a big popup message for everyone might be a downside

**Comment by msgerbs** [comment]
> Is this only visible to the player or to everyone? Things like stealth loading a 357 in your bag showing up a big popup message for everyone might be a downside

Visibility of the messages is exactly the same as before. The only balloon in this PR that shows to other players is the *click* which is consistent with how it was before. The rest are shown only to the user.

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -143,7 +143,7 @@
 	return !user.contains(src)
 
 /obj/item/gun/proc/shoot_with_empty_chamber(mob/living/user as mob|obj)
-	visible_message(span_warning("*click*"), vision_distance = COMBAT_MESSAGE_RANGE)
+	balloon_alert_to_viewers("*click*", vision_distance = COMBAT_MESSAGE_RANGE)
```
Comment (Mothblocks) [inline code/modules/projectiles/gun.dm:146]: Does vision_distance actually work for balloon_alert_to_viewers? I don't think it does.

Comment (Mothblocks) [reply code/modules/projectiles/gun.dm:146]: I think you should just remove the vision distance entirely since there's already a sound for it, and so you'd be able to hear it anyway I'd think.

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -262,11 +262,11 @@
 	if (bolt_type == BOLT_TYPE_OPEN)
 		if(!bolt_locked) //If it's an open bolt, racking again would do nothing
 			if (user)
-				to_chat(user, span_notice("[src]'s [bolt_wording] is already cocked!"))
+				balloon_alert(user, "[bolt_wording] already cocked")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic.dm:265]:
```suggestion
				balloon_alert(user, "[bolt_wording] already cocked!")
```

"!" indicates failure. Also, what is bolt_wording defined as in the usual places? I'm wondering if it fits balloon alert wording rules

Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic.dm:265]: Add "!" everywhere you are indicating failure

File: code/modules/projectiles/guns/ballistic.dm [hunk 3]
```diff
@@ -341,7 +341,7 @@
 			if (tac_reloads)
 				eject_magazine(user, FALSE, AM)
 			else
-				to_chat(user, span_notice("There's already a [magazine_wording] in [src]."))
+				balloon_alert(user, "already loaded")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic.dm:344]:
```suggestion
				balloon_alert(user, "already loaded!")
```

File: code/modules/projectiles/guns/ballistic.dm [hunk 4]
```diff
@@ -485,11 +485,11 @@
 			if(T && is_station_level(T.z))
 				SSblackbox.record_feedback("tally", "station_mess_created", 1, CB.name)
 		if (num_unloaded)
-			to_chat(user, span_notice("You unload [num_unloaded] [cartridge_wording]\s from [src]."))
+			balloon_alert(user, "[num_unloaded] [cartridge_wording] unloaded")
 			playsound(user, eject_sound, eject_sound_volume, eject_sound_vary)
 			update_appearance()
 		else
-			to_chat(user, span_warning("[src] is empty!"))
+			balloon_alert(user, "empty")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic.dm:492]:
```suggestion
			balloon_alert(user, "it's empty!")
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 5]
```diff
@@ -35,11 +35,11 @@
 	if(!select)
 		burst_size = 1
 		fire_delay = 0
-		to_chat(user, span_notice("You switch to semi-automatic."))
+		balloon_alert(user, "semi-automatic")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/automatic.dm:38]:
```suggestion
		balloon_alert(user, "switched to semi-automatic")
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 6]
```diff
@@ -35,11 +35,11 @@
...
 	else
 		burst_size = initial(burst_size)
 		fire_delay = initial(fire_delay)
-		to_chat(user, span_notice("You switch to [burst_size]-round burst."))
+		balloon_alert(user, "[burst_size]-round burst")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/automatic.dm:42]:
```suggestion
		balloon_alert(user, "switched to [burst_size]-round burst")
```

File: code/modules/projectiles/guns/energy/beam_rifle.dm [hunk 7]
```diff
@@ -149,7 +149,7 @@
 
 /obj/item/gun/energy/beam_rifle/attack_self(mob/user)
 	projectile_setting_pierce = !projectile_setting_pierce
-	to_chat(user, span_boldnotice("You set \the [src] to [projectile_setting_pierce? "pierce":"impact"] mode."))
+	balloon_alert(user, "[projectile_setting_pierce? "pierce":"impact"] mode")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/energy/beam_rifle.dm:152]:
```suggestion
	balloon_alert(user, "switched to [projectile_setting_pierce ? "pierce":"impact"] mode")
```

File: code/modules/projectiles/guns/special/syringe_gun.dm [hunk 8]
```diff
@@ -54,7 +54,7 @@
 
 /obj/item/gun/syringe/attack_self(mob/living/user)
 	if(!syringes.len)
-		to_chat(user, span_warning("[src] is empty!"))
+		balloon_alert(user, "empty")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/special/syringe_gun.dm:57]:
```suggestion
		balloon_alert(user, "it's empty!")
```

File: code/modules/projectiles/guns/special/syringe_gun.dm [hunk 9]
```diff
@@ -64,7 +64,7 @@
 	user.put_in_hands(S)
 
 	syringes.Remove(S)
-	to_chat(user, span_notice("You unload [S] from \the [src]."))
+	balloon_alert(user, "[S.name] unloaded")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/special/syringe_gun.dm:67]:
```suggestion
	balloon_alert(user, "unloaded [S.name]")
```

Comment (Mothblocks) [reply code/modules/projectiles/guns/special/syringe_gun.dm:67]: Just flows better

File: code/modules/projectiles/guns/special/syringe_gun.dm [hunk 10]
```diff
@@ -77,7 +77,7 @@
 		if(syringes.len < max_syringes)
 			if(!user.transferItemToLoc(A, src))
 				return FALSE
-			to_chat(user, span_notice("You load [A] into \the [src]."))
+			balloon_alert(user, "[A.name] loaded")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/special/syringe_gun.dm:80]:
```suggestion
			balloon_alert(user, "loaded [A.name]")
```

File: code/modules/projectiles/guns/special/syringe_gun.dm [hunk 11]
```diff
@@ -159,7 +159,7 @@
 		if(syringes.len < max_syringes)
 			if(!user.transferItemToLoc(D, src))
 				return FALSE
-			to_chat(user, span_notice("You load \the [D] into \the [src]."))
+			balloon_alert(user, "[D.name] loaded")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/special/syringe_gun.dm:162]:
```suggestion
			balloon_alert(user, "loaded [D.name]")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/projectiles/gun.dm:146]: Just noticed @MrMelbert brought this up. Does balloon_alert_to_viewers even support this?  I don't remember adding that. Also, is there a case where you hear the sound but not see the message? Both need to happen at the same time for a11y.

Comment (msgerbs) [reply code/modules/projectiles/gun.dm:146]: Vision distance does work. It defaults to `DEFAULT_MESSAGE_RANGE` (7). I'm fine with making this 7 instead of 3 but you'll have to fight Melbert for it.

Comment (Mothblocks) [reply code/modules/projectiles/gun.dm:146]: I just want to make sure there's no case where you can see it but not hear it or hear it and not see it.

File: code/modules/projectiles/guns/ballistic.dm (hunk 2 above)
Comment (msgerbs) [reply code/modules/projectiles/guns/ballistic.dm:265]: `bolt_wording` is usually like "bolt", "slide", etc. It should fit the wording rules from what I tested.

Also the wording guide says no trailing punctuation, does that not included exclamation marks?

File: code/modules/projectiles/guns/special/syringe_gun.dm (hunk 9 above)
Comment (msgerbs) [reply code/modules/projectiles/guns/special/syringe_gun.dm:67]: For context, this comes out to "unloaded syringe" vs "syringe unloaded". I favor the latter as it's clear the former is a noun and the latter is a verb. I don't think your suggestion flows better, imo anyway. Would love somebody else to weigh in here.

File: code/modules/projectiles/guns/ballistic.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic.dm:265]: I need to update whatever you're reading.

Normal actions have no punctuation.
Failure actions end with `!`
Pending actions (such as before a do_after) end with `...`

File: code/modules/projectiles/guns/special/syringe_gun.dm (hunk 9 above)
Comment (Mothblocks) [reply code/modules/projectiles/guns/special/syringe_gun.dm:67]: It's been 11 minutes and I decided I kinda don't care anymore

File: code/modules/projectiles/guns/ballistic.dm (hunk 2 above)
Comment (msgerbs) [reply code/modules/projectiles/guns/ballistic.dm:265]: This is what I am reading: https://hackmd.io/@tgstation/SJIle5wYd#balloon_alertballoon_alert_to_viewers

If there's more current documentation please point me to it, I couldn't find anything.

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (msgerbs) [reply code/modules/projectiles/gun.dm]: I switched this back to DEFAULT_MESSAGE_RANGE.

**Comment by msgerbs** [comment]
Ok I think that's everything I feel like changing. The rest are either too important for balloons (firing the gun!) or too wordy (modifying guns) or too snowflaky for me to care. Also the explorer pin was giving double notifications, hence why that message was removed entirely.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 12]
```diff
@@ -299,13 +299,13 @@
 		..()
 		return
 	if (!cover_open)
-		to_chat(user, span_warning("[src]'s cover is closed! Open it before trying to remove the magazine!"))
+		balloon_alert(user, "cover must be opened first!")
```
Comment (Mothblocks) [inline code/modules/projectiles/guns/ballistic/automatic.dm:302]: Change these to "open the cover!", which is shorter and more direct (and more common with balloons)

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 13]
```diff
@@ -193,23 +193,6 @@
 		if(1)
 			. += "[initial(icon_state)]_burst"
 
-/obj/item/gun/ballistic/automatic/m90/burst_select()
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/guns/ballistic/automatic.dm:196]: What's behind the removal here?

File: code/modules/projectiles/guns/ballistic.dm [hunk 14]
```diff
@@ -611,7 +611,7 @@ GLOBAL_LIST_INIT(gun_saw_types, typecacheof(list(
 
 /obj/item/gun/ballistic/proc/guncleaning(mob/user, obj/item/A)
 	if(misfire_probability == 0)
-		to_chat(user, span_notice("[src] seems to be already clean of fouling."))
+		balloon_alert(user, "it's already clean!")
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/guns/ballistic.dm:614]:
```suggestion
		balloon_alert(user, "already clean...")
```

**Review by JohnFulpWillard — COMMENTED** [review]
just some nitpicking on some balloon alerts

File: code/modules/projectiles/guns/ballistic.dm [hunk 15]
```diff
@@ -575,10 +575,10 @@ GLOBAL_LIST_INIT(gun_saw_types, typecacheof(list(
 	if(!saw.get_sharpness() || (!is_type_in_typecache(saw, GLOB.gun_saw_types) && saw.tool_behaviour != TOOL_SAW)) //needs to be sharp. Otherwise turned off eswords can cut this.
 		return
 	if(sawn_off)
-		to_chat(user, span_warning("[src] is already shortened!"))
+		balloon_alert(user, "it's already shortened!")
```
[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic.dm:578]: what? this is wrong

File: code/modules/projectiles/guns/ballistic.dm (hunk 14 above)
Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic.dm:614]: this is wrong

File: code/modules/projectiles/guns/ballistic.dm (hunk 15 above)
[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic.dm:578]: `!` is for failure, `...` is for pending

File: code/modules/projectiles/guns/ballistic.dm (hunk 14 above)
Comment (Mothblocks) [reply code/modules/projectiles/guns/ballistic.dm:614]: `!` is for failure, `...` is for pending
